packages/components/addon/modifiers/hds-clipboard.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Modifier from 'ember-modifier';
+import { assert } from '@ember/debug';
+
+export const getTextToCopy = (text) => {
+  let textToCopy;
+
+  if (text) {
+    if (typeof text === 'string') {
+      textToCopy = text;
+    } else if (
+      // context: https://github.com/hashicorp/design-system/pull/1564
+      typeof text === 'number' ||
+      typeof text === 'bigint'
+    ) {
+      textToCopy = text.toString();
+    } else {
+      assert(
+        `\`hds-clipboard\` modifier - \`text\` argument must be a string - provided: ${typeof text}`
+      );
+    }
+  }
+  return textToCopy;
+};
+
+export const getTargetElement = (target) => {
+  let targetElement;
+  if (typeof target === 'string') {
+    targetElement = document.querySelector(target);
+
+    if (!targetElement) {
+      console.error(
+        '`hds-clipboard` modifier - `target` selector provided does not point to an existing DOM node, check your selector string',
+        targetElement
+      );
+      return;
+    }
+  } else if (target instanceof Node && target.nodeType === 1) {
+    targetElement = target;
+  } else {
+    if (target instanceof NodeList) {
+      assert(
+        '`hds-clipboard` modifier - `target` argument must be a string or a DOM node - provided: a list of DOM nodes'
+      );
+    } else {
+      assert(
+        `\`hds-clipboard\` modifier - \`target\` argument must be a string or a DOM node - provided: ${typeof target}`
+      );
+    }
+  }
+  return targetElement;
+};
+
+export const getTextToCopyFromTargetElement = (targetElement) => {
+  let textToCopy;
+  if (targetElement instanceof Node && targetElement.nodeType === 1) {
+    if (
+      targetElement instanceof HTMLInputElement || // targetElement.nodeName === 'INPUT' ||
+      targetElement instanceof HTMLTextAreaElement || // targetElement.nodeName === 'TEXTAREA' ||
+      targetElement instanceof HTMLSelectElement // targetElement.nodeName === 'SELECT'
+    ) {
+      textToCopy = targetElement.value;
```

Comment:
So I think there may be an issue generally with the modifier - that it only captures the value on init (render, really).

To see this in the showcases when the target is a `textarea` or `input`, change the text and then click the copy button and you'll get the original text from the render when you paste.

Reply:
It could also be an issue with wiring up the reactivity (in the demo code) - I haven't gotten far enough to determine the cause.

Reply:
@meirish I have double checked and it's working as expected for me 🤔

https://github.com/hashicorp/design-system/assets/686239/af964e79-2db4-4614-936f-3913c17273cb

were you checking on this page? https://hds-showcase-git-fix-copy-snippet-blur-in-dropdown-hashicorp.vercel.app/components/copy/button
or another one?

Reply:
BTW, I would be surprised if it doesn't work: what the code does is to read the input/textarea/select `value` (as in DOM `[node].value`) whenever the user clicks the button, which means it's done dynamically, not at rendering.

Reply:
Ha, I just debugged a bit more and I think the examples on the masked inputs are passing `@text` arg, not `@target` so they are working as expected.